internal/mempool/v1/tx.go

```diff
@@ -198,3 +199,50 @@ func (txs *TxStore) GetOrSetPeerByTxHash(hash [mempool.TxKeySize]byte, peerID ui
 	wtx.peers[peerID] = struct{}{}
 	return wtx, false
 }
+
+// WrappedTxList implements a thread-safe list of *WrappedTx objects that can be
+// used to build generic transaction indexes in the mempool.
+type WrappedTxList struct {
+	mtx tmsync.RWMutex
+	txs []*WrappedTx
+}
+
+func NewWrappedTxList() *WrappedTxList {
+	return &WrappedTxList{
+		txs: make([]*WrappedTx, 0),
+	}
+}
+
+// InsertAt inserts a reference to a WrappedTx object at index i in the the
+// WrappedTxList keeping the list. A caller wishing to keep the list sorted must
+// know the correct index, i, at which to insert. This can be done via a
+// sort.Search with specific index searching criteria.
+func (wtl *WrappedTxList) InsertAt(i int, wtx *WrappedTx) {
+	wtl.mtx.Lock()
+	defer wtl.mtx.Unlock()
+
+	if i == len(wtl.txs) {
+		// insert at the end
+		wtl.txs = append(wtl.txs, wtx)
+	}
+
+	// Make space for the inserted element by shifting values at the insertion
+	// index up one index.
+	//
+	// NOTE: The call to append does not allocate memory when cap(wtl.txs) > len(wtl.txs).
```

Comment:
we should probably initialize the list with a capacity greater than 0.

Reply:
No need IMO. Why?